Add org.devsk.flatconvert

quentinSoranzoKrebs opened the pull request:
<!-- ⚠️⚠️ Submission pull request MUST be made against the `new-pr` **base branch** ⚠️⚠️ -->

<!-- 💡 Go to the preview tab to click the links below 💡 -->

### Please confirm your submission meets all the criteria

- [x] Please describe the application briefly.  
**FlatConvert** is a simple and user-friendly multimedia conversion tool for GNOME. It allows users to convert video and image files to various formats using FFmpeg and Python Pillow under the hood. It integrates well with GNOME by following modern design patterns with GTK4 and libadwaita.

- [x] The domain used for the application ID is [controlled by the application developer(s)][appid-domain] and the [application id guidelines][appid] are followed.  
Application ID: `org.devsk.flatconvert` — The domain `devsk.org` (and the `qsk` namespace on GNOME GitLab) is controlled by the developer.

- [X] I have read and followed all the [Submission requirements][reqs] and the [Submission guide][reqs2] and I agree to them.

- [x] I am an _author/developer_ of the project.

<!-- 💡 Mention the GitHub usernames of any additional maintainers needed below 💡 -->

Additional maintainers: *None*

<!-- ⚠️⚠️ DO NOT modify anything below this line ⚠️⚠️ -->

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository  
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id  
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements  
[reqs2]: https://docs.flathub.org/docs/for-app-authors/submission

hfiguiere commented on org.devsk.flatconvert.json:
```diff
@@ -0,0 +1,108 @@
+	  "--socket=fallback-x11",
+	  "--filesystem=home",
+	  "--device=dri",
+	  "--socket=x11",
```
it either `wayland` + `fallback-x11` or just `x11`. Not both.

hfiguiere commented:
> * The domain `devsk.org`

There is nothing responding at this end.

hfiguiere commented on org.devsk.flatconvert.json:
```diff
@@ -0,0 +1,108 @@
+                    "url" : "https://gitlab.gnome.org/qsk/flatconvert/-/archive/1.1.0/flatconvert-1.1.0.tar.gz",
+		    "sha256" : "aaeb0d9ea198fd423bdf6fb554a041ad63ddd5b776f02abc43c83c1e71120438"
+		}
+            ],
+            "config-opts" : [
+                "--libdir=lib"
```
```suggestion
```

this is unecessary.

hfiguiere commented on icons/org.devsk.flatconvert.svg:
once again. this file a never referenced nor used.

hfiguiere replied:
this was NEVER addressed.

hfiguiere commented on metainfo/org.devsk.flatconvert.metainfo.xml:
```diff
@@ -0,0 +1,15 @@
+    xmlns="http://www.freedesktop.org/standards/appstream"
+    xmlns:its="http://www.w3.org/2005/11/its"
+    type="desktop">
+  <id>org.devsk.flatconvert.desktop</id>
```
id never ends with `.desktop`

hfiguiere commented on metainfo/org.devsk.flatconvert.metainfo.xml:
```diff
@@ -0,0 +1,15 @@
+  <description its:translate="yes">
+    <p>A file conversion application for the gnome desktop environnement.</p>
+  </description>
+</component>
```
missing required `url`, `content_rating`, `releases`, `screenshots`.

Also `description` should not be a repeat of `summary`.

quentinSoranzoKrebs commented:
> > * The domain `devsk.org`
> 
> There is nothing responding at this end.

hfiguiere reviewed (changes requested):
You are far from having resolved all the issues.

hfiguiere commented:
> Build must be offline.

This was never address. Fix ALL the issue before asking for review.

hfiguiere commented on metainfo/org.devsk.flatconvert.metainfo.xml:
this was NEVER addressed.

hfiguiere commented on org.devsk.flatconvert.svg:
this was NEVER addressed.

hfiguiere commented on screenshots/screenshot1.png:
why are screenshots here? This was never requested.

quentinSoranzoKrebs replied:
✅ Fixed. I’ve removed the local screenshot files from the repository.
Screenshots are now hosted externally as recommended:
https://devsk.org/screenshots/screenshot3.png
Thanks for the feedback!

hfiguiere commented on org.devsk.flatconvert.json:
```diff
@@ -0,0 +1,103 @@
+    "runtime-version" : "48",
+    "sdk" : "org.gnome.Sdk",
+    "command" : "flatconvert",
+    "metadata": {
+    	"file": "metainfo/org.devsk.flatconvert.metainfo.xml"
+    },
```
what documentation said this was valid? IT IS INVALID.

hfiguiere commented on org.devsk.flatconvert.json:
```diff
@@ -0,0 +1,103 @@
+	"--share=ipc",
+	"--filesystem=home",
+	"--device=dri",
+	"--socket=wayland",
```
what happen on an X11 session?

quentinSoranzoKrebs commented:
> > * The domain `devsk.org`
> 
> There is nothing responding at this end.

quentinSoranzoKrebs commented:
I have addressed all the points raised: the offline build has been successfully verified, the devsk.org domain is now active, and the files have been thoroughly cleaned.
I would appreciate it if you could kindly review the pull request again at your earliest convenience.

Thank you very much for your time and consideration.

hfiguiere commented on org.devsk.flatconvert.json:
```diff
@@ -0,0 +1,60 @@
+    ],
+    "add-extensions": {
+        "org.freedesktop.Platform.ffmpeg": {
+            "version": "23.08",
```
this version is incorrect. GNOME 48 is based on 24.08. And please indent things properly. (line below)

quentinSoranzoKrebs replied:
Thank you for the feedback.
I have changed the extension version to `24.08` to match the runtime version, and updated the JSON indentation to 4 spaces.
I didn't notice any other indentation issues, but feel free to point them out if I missed something.

bbhtt commented on org.devsk.flatconvert.json:
```diff
@@ -0,0 +1,60 @@
+        "--env=FLATPAK_ENABLE_FFMPEG=1"
+    ],
+    "add-extensions": {
+        "org.freedesktop.Platform.ffmpeg": {
```
you need add-ld-path too see the example here https://docs.flatpak.org/en/latest/extension.html#loading-existing-extensions

bbhtt replied:
and create the directory

quentinSoranzoKrebs replied:
Thank you for the suggestions.
I have added `add-ld-path` and ensured the required directory is created as per the Flatpak documentation.
All changes are included in commit `805bd8f`.
Please let me know if anything else needs adjustment.

quentinSoranzoKrebs commented:
Hello,
I have implemented the requested changes, including adding `add-ld-path`, specifying the creation of the required directory both in the Flatpak manifest and ensuring it exists in the sandboxed environment of the app, and updating the extension version to `24.08` (see commit `805bd8f`).
Could you please have another look when you have time?
Thank you very much!

bbhtt commented on org.devsk.flatconvert.json:
```diff
@@ -0,0 +1,64 @@
+        "--share=network",
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--filesystem=home",
```
why is home needed? does this not support filechooser portal?

quentinSoranzoKrebs commented:
bot, build

bbhtt commented:
why do these start with `_`?

![image](https://github.com/user-attachments/assets/c6479384-231b-4f73-bdff-b7b1d163177c)

missing icon

![image](https://github.com/user-attachments/assets/7b28f8d4-ac1e-4a0e-854d-8cf991ceb23b)

missing icon + why "test page"? preferences are empty

![image](https://github.com/user-attachments/assets/3dde5e5f-0dee-4252-bfd0-e58e3301e0d7)

what is "Aide" here? it opens the same dialogue as above

![image](https://github.com/user-attachments/assets/60784ce5-7fa2-41c6-9d9f-e770294acaf9)

after conversion where does the file get saved? I see `20:05:24,091 - INFO - convert_thread.py - utilisation de _convert_img pour /run/user/1000/doc/d749b1fd/file_example_JPG_2500kB_converted.png` in the terminal but i can't find it.

bbhtt commented on org.devsk.flatconvert.json:
```diff
@@ -0,0 +1,63 @@
+        "--device=dri",
+        "--socket=wayland",
+        "--talk-name=org.freedesktop.FileManager1",
+        "--env=FLATPAK_ENABLE_FFMPEG=1"
```
why do you need an env var and for what? just use FFmpeg if it exists.

bbhtt commented on org.devsk.flatconvert.json:
```diff
@@ -0,0 +1,63 @@
+    "sdk": "org.gnome.Sdk",
+    "command": "flatconvert",
+    "finish-args": [
+        "--share=network",
```
i can't see a reason why it needs network.
